### DISCUSSION TITLE
Intelliflo VF erroring out with overcurrent error

### DISCUSSION THREAD

**Discussion by thrystan** [tagyoureit/nodejs-poolController#756] [Nixie]
I have a 10 year old intelliflo vf and have connected it. I can see it and run it on schedules, but after a schedule ends the pump locks up with a overcurrent error.  The only way I seem to be able to recover is to go into service mode, and then reset the pump. 
Something else I noticed, and I don't know if this is normal or not. When the pump is not errored out, and I don't have a schedule running, the pump starts and stops for a second or two every minute or so. Is this normal?

**Comment by rstrouse** [comment]
None of that is normal.   Do you have schedules defined on the pump?  There are few things that can cause overcurrent.  This includes failing bearings and too much restriction on the suction side.  Are you running Nixie?

**Comment by thrystan** [comment]
I'm running nixie. The pump runs just fine if it is running on it's own, these issues only show up when it's being controlled by Nixie. 
![image](https://user-images.githubusercontent.com/8936451/236709442-7f6d3737-5fc6-459c-9746-af0ceac14258.png)
![image](https://user-images.githubusercontent.com/8936451/236709473-05e97e6d-d06b-47b5-92c4-599f0f00f998.png)

**Reply by rstrouse** [reply]
Please provide a replay capture from the wiki.  Let’s see what we are sending it and what it is replying to us.

**Comment by thrystan** [comment]
[replay.zip](https://github.com/tagyoureit/nodejs-poolController/files/11416399/replay.zip)

**Comment by tagyoureit** [comment]
I had one of my pumps show this overcurrent condition and it also seemed to only happen when connected via RS485.  I'm not sure what the correlation is, but it ended up being a hardware issue and the drive unit needed to be replaced.  Will see if @rstrouse sees anything in the logs, but I'm doubtful.

**Reply by thrystan** [reply]
Hopefully that doesn't end up being the issue, It looks like you might as well replace the whole pump as replace the drive unit. I'm seeing ~$1,000 to replace it and I could get a whole new pump for not much more than that.

**Comment by rstrouse** [comment]
Lets start by first turning off the MQTT interface you have selected for Home Assistant.  If you are using the crewski integration this is not required or desired.  While MQTT is not expensive after it is connected, it eats up cycles trying to connect.  At this point the broker you have on Home Assistant is refusing connection.

What I am not seeing are any commands whatsoever being sent to the pump.  njsPC is not sending anything to the pump (although I would have expected it to).  Can I get you to capture a replay where you turn on the Pool Circuit.  At that point I would expect it to start communicating with the pump.

**Comment by thrystan** [comment]
[replay (1).zip](https://github.com/tagyoureit/nodejs-poolController/files/11433006/replay.1.zip)
[replay (3).zip](https://github.com/tagyoureit/nodejs-poolController/files/11433008/replay.3.zip)
Here are a couple more logs, the first one I started then went out of service mode. The second I rebooted and then grabbed the log. Neither of them appeared to grab control of the pump.

**Comment by rstrouse** [comment]
Please remove the link to the SPI interface that you have set up for the filter relay.  This is an output not an input and I suspect that you are trying to garner pressure from a sensor on SPI.  Is that correct?

**Comment by thrystan** [comment]
Yes, that is what I assumed it was for.

**Reply by rstrouse** [reply]
No that is for a filter relay.  This allows the filter circuit to be triggered for shared systems.

**Comment by thrystan** [comment]
[replay.zip](https://github.com/tagyoureit/nodejs-poolController/files/11439842/replay.zip)
The end of this log has an example of nixie trying to start the pump and the pump getting the error below.
![PXL_20230510_043515531](https://github.com/tagyoureit/nodejs-poolController/assets/8936451/c7e7d8b4-df06-44ac-a862-58c56d789f68)

**Comment by rstrouse** [comment]
When not in service mode does the display on the pump show `Display Not Active`?

**Reply by thrystan** [reply]
Yes, with or without the alarm led flashing.

**Comment by rstrouse** [comment]
In your replay I do not see any pump commands for the flow the only thing in the replay is a request to get control of the pump.  Can I get you to start the replay then manually turn on the pool circuit.  This should start sending requests for 30gpm to the pump.  In the responses I do see from your replay we are asking the pump to shut off and it is returning that it did so.

The end of that log is not showing that Nixie is attempting to turn anything on.  In fact there is no point in that log where there is a call for the pump to do anything other than disable the control panel when you put it back into auto.  You turned on the pool circuit then promptly put it into service mode which releases Nixie control.  Then when you put it back in auto the pool circuit is no longer on and no circuits are calling for any flow.  Oddly, it is as if the pump decided on its own (feature 1 mode 6 assigned to the pump) that it should be running at 30gpm but it was never commanded to do so.

Interestingly, it does indicate that it is priming at 55gpm when the status is returned.

Do you have manually configured flows for any of the last 4 features on the pump?  These override any requests by external automation.

**Comment by thrystan** [comment]
[replay (5).zip](https://github.com/tagyoureit/nodejs-poolController/files/11458325/replay.5.zip)
Ok, I disabled feature 8 and 9, and then captured this replay. Everything appeared to work as expected this time. I'll see what happens when the schedule is set to start this evening.

**Comment by thrystan** [comment]
[replay (6).zip](https://github.com/tagyoureit/nodejs-poolController/files/11458378/replay.6.zip)
And here is a replay where the pump is turning on for a second or two every 30 seconds or so. It only gets up to ~150 watts before it turns off again, but that seems like a not good thing.

**Reply by rstrouse** [reply]
I’ll have a look.  Perhaps we are not frequenting the vs pump control often enough and we lose control of it for a second.  Do you have any of the feature buttons selected?

**Comment by thrystan** [comment]
No, I don't have any buttons selected. At some point the pump errored out
again before the schedule started.

**Reply by rstrouse** [reply]
During the latest replay were there any errors on the pump?  The weirdest thing is that when the pump drive state is released it seems to revert back to 55gpm and reports blockage.  I assume you have 55gpm set up as the priming flow since that seems to be what the pump reports when it is first started.

So here is the sequence when the pool circuit is switched off.  The message numbers appear on the 3rd column.
338 - njsPC first asks for the status from the pump
339 - The pump returns that it is currently running at 30gpm (~1730rpm) as commanded since the pool circuit was on.
340 - njsPC asks the pump to make sure the pump panel is disabled
341 - The pump responds sure you are in control and I have disabled the control panel
342 - njsPC tells the pump that it should set its drive state to 4 which means idle.
343 - The pump responds sure I am now idle
344 - njsPC tells the pump that it should remain powered on but shut down the motor
345 - The pump responds yes my power is on and my motor is off
346 - njsPC asks the pump for it status
347 - The pump responds but shows that it is currently priming at 55gpm but 0rpm.
![image](https://github.com/tagyoureit/nodejs-poolController/assets/47839015/afe35146-0dc8-4830-8b2f-9d4c5eaef9e5)


Pull njsPC, I made some changes to not just release the drive state but also set the flow to 0gpm when the pump should be off.

**Reply by thrystan** [reply]
I did notice that the web display says the pump is at 55gpm when it is off. Which I believe is the priming flow.
![image](https://github.com/tagyoureit/nodejs-poolController/assets/8936451/8ed25672-f536-4129-9412-e110925dd52d)

**Reply by rstrouse** [reply]
The pump response is actually saying that it is priming.  Just so you know service mode will cause njsPC to stop talking to the pump so at this point the pump may be sending messages but it has been released from command.

**Reply by thrystan** [reply]
Right, until I can get this to work reliably, I'm leaving it in service mode so that the pump will run on it's own scheduler.

**Reply by rstrouse** [reply]
Please let me know if the changes I made turn off the flow on that pump.  What else is odd is that all of the responses from the pump indicate a blockage on the status but that may be because none of the devs have a VF pump and the users that we were working with in the past didn't follow through.  In your status messages byte index 12 is always 30.  When translated given the know intelliflo error codes are:
Filter Warning, Overcurrent, Priming, and System Blocked.  The highest order condition is typically what is reported but if there is nothing on your display at the time I am suspect on what is actually reporting on that byte.  I have a couple of other replays from VF pumps so I will verify against those.

**Reply by thrystan** [reply]
Will do, I'm working at the moment, but I'll work on this after I get done which should be in a couple of hours.

**Reply by rstrouse** [reply]
Also please go into the schedule configuration and press Save Schedule for each of your schedules.  Oddly, it is not showing the correct schedule times in the state.

**Comment by thrystan** [comment]
How do I pull njsPC?

**Comment by thrystan** [comment]
Ok, I did that. I was wondering why they stopped showing the schedule info
on the main page.

**Reply by rstrouse** [reply]
Run the following commands

`cd ~/nodejs-poolController`
`git pull`
`npm i`

**Comment by thrystan** [comment]
Should I have both a njsPC and a poolController pm2 job running?

**Reply by rstrouse** [reply]
It doesn't matter.  pm2 should restart with the new code when you pull.

**Comment by thrystan** [comment]
[replay (1).zip](https://github.com/tagyoureit/nodejs-poolController/files/11468478/replay.1.zip)
[replay (2).zip](https://github.com/tagyoureit/nodejs-poolController/files/11468479/replay.2.zip)
That doesn't seem to have worked. I'm no longer seeing the watts change when the pump runs briefly, but it's still doing it. And this time it started and was running up fairly fast, without me asking it to start and without showing the speed or watts being used. 
Now when I tell it to run it shows the speed and watts as expected, but not when it does whatever it does that causes it to run on its own. On the plus side it appears to be clearing the error and starting the pump when asked. Also when not running anything it is showing 55gpm on the main page.

**Reply by rstrouse** [reply]
I will dive into the comms this weekend but this really looks like there is a conflict between the schedules on the pump and manual control.  I did notice that your longitude is not correct.  It should be -121.2358.

**Comment by thrystan** [comment]
Ok, thanks.

**Reply by rstrouse** [reply]
Did you get any errors when you updated?  Do this again and pay close attention to the `git pull` command.
`cd ~/nodejs-poolController`
`git pull`
`npm i`

**Reply by thrystan** [reply]
I got an error saying that I needed to commit or stash the 2 package files. I moved them to a temporary directory and then moved them back after the pull. Then ran the npm i. There weren't any errors.

**Reply by rstrouse** [reply]
Ok run the following commands.  You weren't actually updating from the repo.
`cd ~/nodejs-poolController`
`git reset --hard`
`git pull`
`npm i`

**Reply by thrystan** [reply]
It says already up to date.

**Reply by rstrouse** [reply]
Ok now lets do that replay.  Before the pulls were not coming through because git stopped it from pulling the changes I made for you.

**Comment by thrystan** [comment]
Now, I can't connect to njsPC.

**Reply by rstrouse** [reply]
Double check the IP address and verify that it is running by typeing `pm2 monit`

**Comment by thrystan** [comment]
![image](https://github.com/tagyoureit/nodejs-poolController/assets/8936451/fe4cac69-d03b-4756-b95c-edc36b56dc09)
![image](https://github.com/tagyoureit/nodejs-poolController/assets/8936451/4b92ace8-0d02-4d30-bcb4-4536e1b134bf)

**Comment by rstrouse** [comment]
Run the following commands
`pm2 stop all`
`nvm alias default 18`
`cd ~/nodejs-poolController`
`npm rebuild`
`pm2 restart REM`
`pm2 restart dashPanel`
`pm2 restart njsPC`

Then verify that njsPC started.

**Comment by thrystan** [comment]
[replay (8).zip](https://github.com/tagyoureit/nodejs-poolController/files/11470615/replay.8.zip)

**Reply by rstrouse** [reply]
Ok so now it looks like we are on the current version and I am seeing a sequence that is not starting our pump.  The poll is suspended on the pump.
```
[5/13/2023, 2:18:18 PM] info: [2:18:18 PM] 192.168.0.192 GET /state/all?null {}
[5/13/2023, 2:18:21 PM] info: [2:18:21 PM] 192.168.0.192 GET /state/pump/50?null {}
[5/13/2023, 2:18:22 PM] info: [2:18:22 PM] 192.168.0.192 PUT /state/circuit/setState {"id":6,"state":true}
[5/13/2023, 2:18:22 PM] verbose: Turning on a body circuit Pool
[5/13/2023, 2:18:22 PM] info: NCP: Setting Circuit Pool to true
[5/13/2023, 2:18:22 PM] info: Pump 50 Polling Suspended
[5/13/2023, 2:18:24 PM] warn: Inactivity timeout for 0 serial port /dev/ttyUSB0 after 10 seconds
[5/13/2023, 2:18:24 PM] info: Serial Port 0 - /dev/ttyUSB0 has been closed. 
[5/13/2023, 2:18:24 PM] info: Successfully closed portId 0 for serial port /dev/ttyUSB0
[5/13/2023, 2:18:24 PM] info: Serial port: /dev/ttyUSB0 request to open successful 9600b 8-none-1
```

**Reply by rstrouse** [reply]
Well I figured out a couple of things.  Our capture replay is at fault here.  If you capture the replay on a Nixie controller and tell it to reload the configuration it attempts to reconstitute the state.  Since Nixie does not have an OCP to respond to the state right away it clears the schedules and shuts down the pump polling.  I am working on rectifying this as I type this but I suspect that when you capture your replay you are reloading the configuration or reloading the configuration from the button on the hamburger menu.

**Reply by rstrouse** [reply]
Can I get you to restart the pi and capture a replay where you do not check the "Capture Configuration Reload" checkbox.  This is not required or desired with a Nixie controller since there is nothing that the OCP can tell us that we don't already know about the configuration.

**Comment by thrystan** [comment]
Rplay 9 was running when it started up, I ended it and then captured replay
10

**Reply by rstrouse** [reply]
Yeah for Nixie controllers capturing a replay with the reload configuration option results in bad results and the state doesn't actually reset.  This is why we are not getting pump responses.  So stop the replay and completely restart the pi.  Then capture the replay without checking the box.

**Comment by thrystan** [comment]
I did that and that is what is in replay 10

**Reply by rstrouse** [reply]
But that never made it here.  What am I missing?

**Comment by thrystan** [comment]
[replay (10).zip](https://github.com/tagyoureit/nodejs-poolController/files/11473073/replay.10.zip)
Sorry, I didn't notice that it hadn't uploaded.

**Comment by rstrouse** [comment]
Pull njsPC.  It should now put the state data back to where it should be.  I want to get a handle on what our symptoms currently are.  From the replay 10.  I see the conversation working as expected with one exception.  When njsPC is asking the pump to stop it is reporting that it is running at 55gpm.  See the sequence below.

![image](https://github.com/tagyoureit/nodejs-poolController/assets/47839015/f3514ece-3850-4994-a917-08363417e94e)

Message 14 is njsPC asking the pump to set its flow to 0.  That is the last two bytes of the payload.  But in a weird twist it is responding with a 255 and a drive state of 10.  Typically this indicates that it cannot be turned off because a feature has it on.  Update njsPC, I have made the system preserve the state if the controller gets reloaded and have made the drive state remain because if you look at the sequence below.  njsPC is requesting that the pump run at 30gpm on message 279.  The pump promptly responds with 0,30 indicating that it is targeting our requested speed.   Then on message 284 you can see that it is reporting 30gpm just like we asked it to.  So I retained the drive state for the pump in the latest software.

![image](https://github.com/tagyoureit/nodejs-poolController/assets/47839015/50ae874a-0e45-4214-9f69-b0668998b5ef)


After pulling the new code please go back through and point out the things that it is doing correctly and those that it is not.  Perhaps this release of drive state represents the surging.  It is possible that when the speeds and schedules are set on the VF pump they override the requested speed when the drive state is released with a 4 instead of a 10.  I don't want you to reset the pump drive to factory settings I want to see if we can get it to respond properly with a drive state that cannot be set from the menus.

**Comment by thrystan** [comment]
[replay (11).zip](https://github.com/tagyoureit/nodejs-poolController/files/11473248/replay.11.zip)
Ok, things appear to work, except that it started the pump at 30 gpm, without a schedule being set to run, and I can't get it to stop the pump. When I selected the surface sweep feature, it successfully switched to 60 gpm, but when I turned it off, it remained at 60 gpm until I turned the 30 gpm feature on, and when I turned that off it remained running at 30 gpm. So basically it appears to not return to a 0 gpm state when the pump should be off.

Also something I noticed, is when I turn off the Ras Pi, or disconnect the pump, it returns to a state where feature 1 is running, not the main programs running state that I would expect.

**Reply by rstrouse** [reply]
Ok I am sure that this is because there are programs on feature 1 and 4.  I added a VF pump to my IntelliCenter (although I do not have a VF pump) and it sent the exact messages to the pump that we are sending from Nixie.

**Reply by thrystan** [reply]
So, should I clear feature 1? Or do you want to see if you can come up with something to fix it without touching the features?

Also what about it not stopping the pump when a feature, or schedule ends?

**Reply by rstrouse** [reply]
The not stopping is likely because of the speed set on the features on the pump.  Where I have seen folks switching to automation all 8 features were disabled on the pump.

**Reply by thrystan** [reply]
Ok, I'll try clearing the features on the pump.

**Reply by thrystan** [reply]
Apparently you can't clear or disable features 1 and 2 on the pump. When you say programs are you talking about the number of cycles and turns per day that are programmed?

**Comment by thrystan** [comment]
[replay (13).zip](https://github.com/tagyoureit/nodejs-poolController/files/11474113/replay.13.zip)
Also the schedules don't seem to be triggering. This is a replay from a minute before to a minute after the main schedule is supposed to start.

**Reply by rstrouse** [reply]
The schedules issue is because you have the start time after the end time.  This is in parity with the way Pentair panels work but there really doesn't need to be this restriction for Nixie.  We just need to code around the Pentair panels.

**Reply by thrystan** [reply]
Ok, I wasn't sure if it would work or not. I'll switch to two programs one for before midnight and one afterwards.

**Reply by rstrouse** [reply]
If you want part of your pool schedule you can simply overlap that into the pool schedule.  More than one schedule can run at a time.

**Reply by thrystan** [reply]
The issue is I'm on Time of Use electric rates where the rates are higher from 4-9 so I want the pump running from 21:30 until 15:30. But if having the start time later than the end time doesn't work, which I can totally understand being the case. I'm a programmer who works on time card systems among other things, and dayend issues are a PITA. Anyway I can easily just have a 21:30-23:59 and a 0:00 to 15:30 schedule. But the front end let me enter it as just one so I figured I'd see if it would work.

**Reply by rstrouse** [reply]
Yeah the expectation was to code around the limit.  It just got missed.   The Pentair panels would have failed at the server.

**Reply by thrystan** [reply]
Things seem to be working except it doesn't shut down when I tell it to.

**Comment by rstrouse** [comment]
Update your njsPC.  This should shut down the pump when there is no flow.  The drive state settings are now being done at a faster interval.

**Reply by thrystan** [reply]
Excellent, that seems to be working. The pump still says 55gpm when stopped, but the watts is 0. I'm not sure if you want to keep poking at that bit or not. If not I think this should be closed. I've got another question or two, but I'll open separate discussions for that.

**Reply by rstrouse** [reply]
One more replay please.  Part of the issue is that while there have been other users with IntelliFlo VF we never got a baseline where it was all working.  I suspect the 55 is a the priming flow and is probably reported because of settings on the pump.  I may go in and check the drive state on the status return.

**Reply by thrystan** [reply]
[replay (14).zip](https://github.com/tagyoureit/nodejs-poolController/files/11490795/replay.14.zip)
And now it's not turning off again, actually nothing is changing now.

**Reply by rstrouse** [reply]
Restart the pi.  I don't know why the polling is suspended on the pump.  Was it working before you started the replay?

**Reply by tagyoureit** [reply]
> One more replay please. Part of the issue is that while there have been other users with IntelliFlo VF we never got a baseline where it was all working. I suspect the 55 is a the priming flow and is probably reported because of settings on the pump. I may go in and check the drive state on the status return.

I did a lot of work on the VF pumps (and some replays) here: https://github.com/tagyoureit/nodejs-poolController/issues/355

**Reply by thrystan** [reply]
That thread explains why it was always ending up running feature 1 when I would stop njsPC. 

I've been watching this software on and off for several years, at first just because I wanted the protocol so I could write something for myself. Then when I finally got around to building the first version of my pool controller 3 years ago I installed the software, but I couldn't get the rs485 connection to work with either the max485 module I had or the usb interface I bought. So I just bypassed my need for it by installing a flow meter, because the only thing I really needed was to know if the pump was running so I could know if it was safe to pump chlorine or acid. A couple versions of the perfboard hardware later I finally switched to a PCB and tried to connect again, and once I bought the current recommended usb interface it worked. 

 I'm a programmer for my day job and I do enough software support there that I don't want to do support as a hobby. So I really appreciate all the work you guys have done to write and more importantly maintain this. My pool is simple enough that I don't need most of the features, it's a 70's pool with a single pump and no heater or other features, I replaced the original intermatic timers with the inteliflo vf and it ran on it's own for the last 10 years with no problems, but I am a nerd at heart and having more things connected and visible is always better. :) And for some of the modern setups I see on the troublefreepool forums, your software is amazing.

Currently I'm waiting on the 3rd version of the pcb for my controller and I already know that I'm going to be getting at least one more board, that will hopefully have integrated rs485, along with the current analog inputs, 1 wire thermometer inputs, i2c connectors, and chlorine and acid pump controllers. And this software is what will allow me to share it with some of my friends with pools.

**Reply by thrystan** [reply]
[replay (16).zip](https://github.com/tagyoureit/nodejs-poolController/files/11494492/replay.16.zip)
It looks like sometime between when the schedule stopped the pump at 3:30 and 9:30 when it's supposed to start again, something happened similar to what happened the last time. At any rate the pump isn't responding.

**Reply by thrystan** [reply]
The pump had errored out again.

**Reply by rstrouse** [reply]
@tagyoureit the one thing that is missing from all those massive replays is when the pump is supposed to turn off.  In @thrystan's case when the drive state changes to 4 it shows 55gpm and seems to get stuck.

@thrystan I assume that the replay above (replay (16).zip) is after the pump is stuck.  Here is what we are seeing at this time.  You should be able to see these messages in real time using [Message Manager](https://github.com/rstrouse/nodejs-poolController-dashPanel/wiki/Message-Manager).  No need to install anything it is already installed with dashPanel.

Anyhow here is the sequence.
![image](https://github.com/tagyoureit/nodejs-poolController/assets/47839015/9643d47f-e362-497b-bbf8-a8576d8e0ac9)

1. On message 317 Nixie tells the pump to disable the control panel and on 318 the pump responds with Ok I am disabled.
2. On Message 319 Nixie tells the pump that we need a drive state of 10.  10 means that we should be running a drive state of 4 means stop.  Then on message 320 the pump says Ok you have a drive state of 10.
3. On message 321 Nixie tells the pump to turn on feature 1 (6 means feature 1) and obediently the pump responds I set to feature 1 with message 322.
4. On message 323 Nixie tells the pump that it should set the flow to 30.  The first 2 bytes are a constant and the last two are the flow.  The pump then responds on 324 that the flow was accepted.
5. On message 325 Nixie asks the pump to tell it what it is currently doing.  And here is where it gets interesting on message 326 the pump responds with the following bytes that I will lay out for you.  The discrepancy here is that is seems to be stuck priming but I would have expected it to have rpm and power.

|  Index | Value | Name | Description |
| --- | --- | --- | --- |
| 0 | 10 | Command | This is a bitmask where bit 2 indicates that the power is on and bit 4 indicates that it is running |
| 1 | 6 | Mode | Feature 1 is selected. |
| 2 | 1 | State | 0 = Fault, 1 = Priming, 2 = Running.  In this case the pump says it is priming |
| 3 - 4 | 0, 0 | Power | The power in watts.  In this case the pump says 0 watts |
| 5 - 6 | 0, 0 | Speed | The speed in rpm.  In this case the pump say it is running at 0 rpm. |
| 7 | 55 | Flow | The flow in GPM.  In this case the flow is 55gpm (yet 0 watts and 0 rpm) |
| 8 | 0 | PPC | The power curve for the motor.  This too is 0 but that is often fine. |
| 9 | 1 | Unknown | We really have no idea what this byte is for but it often oscillates between 0 and 1 under normal operation |
| 10 | 0 | Unknown | We really have no idea what this byte means but it is often 0 under normal operation |
| 11 - 12 | 0, 1 | Alarms | This is a bitmask of any alarms from the pump.  In this case 1 means Ok.  The alarm bits are listed below at one point we were seeing 0, 30 which indicates a priming error caused by an overcurrent.  The filter and system blocked bits were also set as if it had restriction.  Basically, all kinds of issues |
| 13 - 14 | 22,28 | Runtime | Pump runtime used to calculate filter cleaning and the like. |


Alarm Bits
```
1 = 0k
2 = Filter Warning
3 = Drive Overcurrent
4 = Priming Error
5 = System Blocked
6 = Drive Error
7 = Overtemp
8 = Power outage/brownout
9 = Motor Overcurrent
10 = Overvoltage
```

**Reply by thrystan** [reply]
yes replay 16 was when the pump was not responding. I know that when I went out to the pump, the alarm led was flashing, and when I put it in service mode, and then cleared the alarm, and put it back in auto everything started working again. 
I understand the history for setting the pump to feature 1, But I don't really understand why that is done, other than it worked.
I'm wondering if the overcurrent issue could be caused by the stop of the midnight to 3pm schedule and the immediate start of the 3pm to 3:30pm schedule. Could the stop/start cause an issue, I know with other industrial motors quick stops and starts can cause overcurrent issues due to the inertia of starting and stopping the attached mass, so they usually try to do some kind of ramp up /ramp down. I switched the schedule so that the main pool schedule now runs from midnight to 3:45pm and the surface sweep higher gpm schedule just runs on top of that from 3:00 to 3:30. Maybe that will help?

**Reply by rstrouse** [reply]
I would set it up so that the flow just increases for the sweep.  It is so odd that the pump thinks that it is stuck priming but actually doing nothing and not reporting any alarms on the status message.  

The feature selection comes from the way Pentair panels interact with this pump.  If you do not select a feature it will not run.  This is true only for VF and VS+SVRS pumps and not for VS and VSF pumps.

**Reply by thrystan** [reply]
Ok, I'll see what happens when it shuts down in about an hour.
I'm excited because my new PCBs just arrived so I'll be playing with them this evening.
![PXL_20230517_212131016](https://github.com/tagyoureit/nodejs-poolController/assets/8936451/736dff3f-9ada-46c0-916b-2595227268f6)
![PXL_20230517_211905054](https://github.com/tagyoureit/nodejs-poolController/assets/8936451/c7bb319f-ce46-4c65-8011-fa93269b273a)
This isn't my final version I've already found some changes I'm going to make, but I need to test what I changed on this version first.

**Reply by rstrouse** [reply]
Well that looks like you have a date with a soldering iron.

**Reply by thrystan** [reply]
Yep, I've got a hot date tonight.

**Reply by rstrouse** [reply]
Out of curiosity why did you choose 1-wire instead of 10k NTCs?  The install is much simpler, they are reliable, accurate, and readily available.

**Reply by thrystan** [reply]
I had them, so I didn't look for anything else.

**Reply by thrystan** [reply]
![PXL_20230518_041501490](https://github.com/tagyoureit/nodejs-poolController/assets/8936451/a18cb01a-3816-46bc-bed1-d96231fcc8ec)
![PXL_20230518_041510099](https://github.com/tagyoureit/nodejs-poolController/assets/8936451/257cfb69-348e-4358-8035-92fd899c98e5)
![PXL_20230518_050405242](https://github.com/tagyoureit/nodejs-poolController/assets/8936451/3b070f97-1b49-4224-a56f-37ba2e8aa996)
 My soldering iron and I had a wonderful evening. This is the first time I've done smd chips. With a hot air pencil they aren't bad

**Reply by rstrouse** [reply]
Did you order dessert.  It looks like it was a fine cheese and granola type of night.  That looks awesome are you planning any pH or ORP sensors?

**Reply by thrystan** [reply]
Yes, I have ph and orp sensors, they are on another board that is closer to the plumbing. The probe leads aren't long enough to get to the box with this board. I did a post about my pool story on trouble free pools a month or so ago with lots of pics in the everything else forum.

**Comment by thrystan** [comment]
It had been working, I didn't check right before the replay though so I can't say for sure that it stopped working when I started the replay.
[replay (15).zip](https://github.com/tagyoureit/nodejs-poolController/files/11491027/replay.15.zip)
rebooting fixed it.

**Reply by thrystan** [reply]
I'm still having issues with the pump starting and stopping with an alarm light every 20-30 seconds when the pump should just be off.
[replay (20).zip](https://github.com/tagyoureit/nodejs-poolController/files/11509081/replay.20.zip)
This reply captures about 3 of these surges.

**Reply by rstrouse** [reply]
The replay indicates that the pump is attempting to prime.  These are the surges.  I have a couple of questions
1. Is the red start/stop light illuminated when it is doing this?
2. Do you have a system priming time set?
3. Are any of the program settings enabled?

After you have verified this pull njsPC and lets see if setting the feature to undefined causes it to not start a priming cycle.

**Reply by thrystan** [reply]
I did notice the start/stop led (it's amber not red) turn on when the pump started.
I'm not sure what you mean by system priming time. I don't have anything that I know of set where it just primes the pump at certain times.
There is a program setup to run 1 turn a day for 30000 gallons with two cycles scheduled for 0:00 to 15:30 and 21:30 to 23:59. But when the surges happen the feature 1 led is on not the filter mode led.

**Reply by rstrouse** [reply]
Page 15.
https://www.pentair.com/content/dam/extranet/nam/product-related/product-manuals/pool-and-spa-equipment/intelliflo-vf/IntelliFlo_VF_OM.pdf

Please pull njsPC and shoot me a replay when the pump is supposed to be off.  I set the feature to undefined so I hope it turns off feature 1 altogether when it is supposed to shut down.

**Reply by thrystan** [reply]
Ahh, that yes I have it set to 5 min I believe. I'll load the replay shortly.

**Reply by thrystan** [reply]
That appears to have fixed the surging. 
[replay (21).zip](https://github.com/tagyoureit/nodejs-poolController/files/11510551/replay.21.zip)
When I turn it off it stays off.

**Comment by rstrouse** [comment]
Is there anything left still outstanding with this?

**Reply by thrystan** [reply]
I don't believe so. I think it should be good now.

**Reply by rstrouse** [reply]
Awesome thanks for sticking with this.  We don't have this equipment and sometimes we need to throw some spaghetti at it.

**Reply by thrystan** [reply]
Thank you for all the time. Support like this is one of the main reasons I've never distributed any software.
